### PR TITLE
Fix step walker and move to fort issues

### LINKED ISSUE #5384: Bot goes back and forth between normal and lured forts
### Expected Behavior
Bot moves normally between forts.

### Actual Behavior
It goes back and forth between a normal and a lured fort.
http://imgur.com/WZSOEbR

### Your FULL config.json (remove your username, password, gmapkey and any other private info)
http://pastebin.com/gUUT6fiA

### Output when issue occurred

> [2016-09-11 10:10:05] [MoveToFort] [INFO] Moving towards pokestop Roman Boy Holding Paper - 0.07km
> [2016-09-11 10:10:27] [MoveToFort] [INFO] Moving towards pokestop Roman Boy Holding Paper - 0.08km
> [2016-09-11 10:10:49] [MoveToFort] [INFO] Moving towards pokestop Памятник перед муниципальным о - 0.07km (attraction of
>  lure 0.07km)
> [2016-09-11 10:10:53] [IncubateEggs] [INFO] Eggs incubating: [4.23/5.0 km] (Eggs left: 8, Incubating: 1)
> [2016-09-11 10:11:09] [MoveToFort] [INFO] Moving towards pokestop Памятник перед муниципальным о - 0.07km (attraction of
>  lure 0.07km)
> [2016-09-11 10:11:30] [MoveToFort] [INFO] Moving towards pokestop Roman Boy Holding Paper - 0.08km
> [2016-09-11 10:11:51] [MoveToFort] [INFO] Moving towards pokestop Памятник перед муниципальным о - 0.07km (attraction of
>  lure 0.07km)
> [2016-09-11 10:12:13] [MoveToFort] [INFO] Moving towards pokestop Roman Boy Holding Paper - 0.07km
> [2016-09-11 10:12:34] [MoveToFort] [INFO] Moving towards pokestop Roman Boy Holding Paper - 0.06km
> [2016-09-11 10:12:55] [IncubateEggs] [INFO] Eggs incubating: [4.23/5.0 km] (Eggs left: 8, Incubating: 1)
> [2016-09-11 10:12:55] [MoveToFort] [INFO] Moving towards pokestop Памятник перед муниципальным о - 0.07km (attraction of
>  lure 0.07km)
> [2016-09-11 10:13:19] [MoveToFort] [INFO] Moving towards pokestop Памятник перед муниципальным о - 0.07km (attraction of
>  lure 0.07km)
> [2016-09-11 10:13:40] [MoveToFort] [INFO] Moving towards pokestop Памятник перед муниципальным о - 0.07km (attraction of
>  lure 0.07km)
> [2016-09-11 10:14:01] [MoveToFort] [INFO] Moving towards pokestop Roman Boy Holding Paper - 0.08km

### Other Information
OS: 
Arch Linux ARM
Branch: 
master
Git Commit: 
11123a82cb18c172a262dc9e46e7b2c175666828
Python Version: 
2.7.12

### PR DESCRIPTION
**Pull request by Anakin5**
- Fix step walker timing issue. This solve some problems where we were moving more than the configured speed from time to time.
- Fix move to fort task infinite loop with Polyline. Because MoveToFort was always taking the nearest fort, and because we do not always choose the straight line to move (Polyline), you might end up in a scenario where you infinitely switch between 2 forts.
- Improve a bit the computation speed of step_walker get_next_step
- pep8 fixes

Will fix #5384 but potentially other weird behavior as well

### CODE REVIEW

File: pokemongo_bot/cell_workers/move_to_fort.py [hunk 1]
```diff
@@ -38,79 +43,57 @@ def work(self):
...
 
-        nearest_fort = self.get_nearest_fort()
+        if self.destination is None:
+            self.destination = self.get_nearest_fort()
 
-        if nearest_fort is None:
-            return WorkerResult.SUCCESS
+            if self.destination is not None:
+                self.walker = walker_factory(self.config_walker, self.bot, self.destination['latitude'], self.destination['longitude'])
+                self.walker.is_arrived = self.walker_is_arrived
```
Comment (th3w4y) [inline pokemongo_bot/cell_workers/move_to_fort.py]: This looks fishy...

Comment (th3w4y) [inline pokemongo_bot/cell_workers/move_to_fort.py]: You are patching walker is arrived? 👍

**Comment by th3w4y** [comment]
We would be probably better off by adding a parameter to the step walker __init__ if you want to modify the StepWalker behaviour...

**Comment by th3w4y** [comment]
@Anakin5 is this still a change you want to do? I see it now detached from your repo/branch

**Comment by Anakin5** [comment]
it got detached when this repo has been deleted. I still plan to do the changes and create another pr

**Comment by davidakachaos** [comment]
@Anakin5 ping?

**Comment by MerlionRock** [comment]
@davidakachaos Do you mind merging this to your changes? So that when both yours and this are committed, I can move on to investigate what causes walker not moving after a session slate.

**Comment by davidakachaos** [comment]
@MerlionRock I don't mind, but my two PRs were merged already. And I didn't modify these files, so could you please tell me where I should merge?

**Comment by MerlionRock** [comment]
@davidachaos then maybe I'll try to raise a PR myself base on what has already being merged.

Just not sure how to correct that Monkey patching....

**Comment by davidakachaos** [comment]
@MerlionRock Okay, well, if I can help you out in some way, let me know :smile:

**Comment by MerlionRock** [comment]
@davidakachaos Step walker has no conflict, I'll do a PR for it.

I've changed move_to_fort with a little simple logic to prevent it from infinitely switching between 2 forts instead. Will merge it instead of using Anakin5's.

But I'll wait for the 0.53 API to be ready first before doing these PR.

**Comment by davidakachaos** [comment]
Cool. Now we wait for a stable api again. Hope it will be available soon.

Op 21 jan. 2017 04:56 schreef "MerlionRock" <notifications@github.com>:

**Comment by davidakachaos** [comment]
@MerlionRock I think this PR can be closed now? cc @solderzzc

**Comment by MerlionRock** [comment]
@davidakachaos I guess so. If @Anakin5 ever return, he can reopen it. cc @solderzzc

**Comment by solderzzc** [comment]
Yes, Let me close it by now.  @davidakachaos  @MerlionRock

**Pull request closed without merging**